### PR TITLE
fix for #23. Add support for pfx/p12 certificate outputs

### LINKED ISSUE #23: Add support for pfx/p12 certificate outputs
**BUSINESS PROBLEM**
At the moment the Venafi terraform provider only outputs PEM formatted certificates and private keys. This is fine for AWS based deployments, but Azure requires pfx formatted certificates to natively integrate with each Azure service.

**PROPOSED SOLUTION**
Natively output pfx formatted certificates from the Venafi provider. This would allow us to easily interact and inject the pfx certificate in to Azure Key Vault so it can be consumed by different Azure services natively. 

The solution needs to be ingestible by the Azure Key Vault (https://www.terraform.io/docs/providers/azurerm/r/key_vault_certificate.html#example-usage-importing-a-pfx-) contents parameter: 
```
certificate {
    contents = filebase64("certificate-to-import.pfx")
    password = ""
  }
```

**CURRENT ALTERNATIVES**
Write some custom code to convert the PEM formatted certificates and keys to pfx.

### PR DESCRIPTION
**Pull request by angelmoo**
Added a new object to the output to support  pfx/p12.

added the pkcs12 and with this user can call it on the terraform configuration file, for example:

output "pkcs12_certificate" {
   value = "${venafi_certificate.webserver.pkcs12}"
}

and then he will obtain the value which is a PKCS#12 keystore containing certificate, private key, and chain certificates that is encrypted using the password value specified by the key_password argument

Ran test suit and All test cases passed.

### CODE REVIEW

**Review by tr1ck3r — APPROVED** [review]

File: venafi/resource_venafi_certificate.go [hunk 1]
```diff
@@ -356,5 +364,76 @@ func enrollVenafiCertificate(d *schema.ResourceData, cl endpoint.Connector) erro
...
+			return fmt.Errorf("error setting pkcs12: %s", err)
+		}
+	}
+
 	return d.Set("private_key_pem", pcc.PrivateKey)
 }
+
+func AsPKCS12(certificate string, privateKey string, chain []string, keyPassword string) ([]byte, error) {
+
+	if len(certificate) == 0 || len(privateKey) == 0 {
```
Comment (asquith-venafi) [inline venafi/resource_venafi_certificate.go:383]: empty string in Go is can also be done with testing empty string afaik (but this way - len(string) == 0) is not incorrect)
if certificate == "" || privateKey == ""

**Review by asquith-venafi — APPROVED** [review]
